src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs

```diff
@@ -171,22 +171,40 @@ private static object Locale_Parser(ReadOnlyMemory<char> name, PatternTable tabl
                 }
             }
 
-            if (name.Length == 2)
+            // We use a heuristic here for common locale codes. Locale codes are often
+            // * two characters for the language: en, es, fr, de
+            // * three characters for the language: agq
+            if (name.Length == 2 || name.Length == 3)
```

Comment:
This length=3 check wasn't discussed in the issue but is required to pass tests because all of the XXX-YYYY locales also have a XXX form.. Hopefully it's not taking thing a bit too far.